Python/codegen.c

```diff
@@ -6121,7 +6121,9 @@ codegen_match_inner(compiler *c, stmt_ty s, pattern_context *pc)
         }
         // Success! Pop the subject off, we're done with it:
         if (i != cases - has_default - 1) {
-            ADDOP(c, LOC(m->pattern), POP_TOP);
+            /* Use the body location to give better locations for branch events */
+            assert(asdl_seq_LEN(m->body) > 0);
+            ADDOP(c, LOC(asdl_seq_GET(m->body, 0)), POP_TOP);
```

Comment:
Shouldn't this be the location of the last instruction in the block? Line tracing might look like we executed that line more than once if we "return" to it after running another line.

Reply:
I've added a `NEXT_LOCATION` pseudo location so that the `POP_TOP` gets the location of whatever follows it.